extensions/amp-date-picker/0.1/test/test-amp-date-picker.js

```diff
@@ -0,0 +1,171 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import '../../../../third_party/react-dates/bundle';
+import {AmpDatePicker} from '../amp-date-picker';
+import {requireExternal} from '../../../../src/module';
+import {toggleExperiment} from '../../../../src/experiments';
+
+describes.realWin('amp-date-picker', {
+  amp: {
+    runtimeOn: true,
+    extensions: ['amp-date-picker'],
+  },
+}, env => {
+  const moment = requireExternal('moment');
+  let win;
+  let document;
+
+  const templateJson = `{
+    "templates": [{
+      "id": "srcTemplate",
+      "dates": [
+        "2018-01-01",
+        "FREQ=WEEKLY;DTSTART=20180101T080000Z;WKST=SU;BYDAY=WE"
+      ]
+    }, {
+      "id": "defaultTemplate"
+    }]
+  }`;
+
+  const DEFAULT_ATTRS = {
+    layout: 'fixed-height',
+    height: '360',
+  };
+
+  function createDatePicker(opt_attrs) {
+    const element = document.createElement('amp-date-picker');
+    const attrs = Object.assign({}, DEFAULT_ATTRS, opt_attrs);
+    for (const key in attrs) {
+      element.setAttribute(key, attrs[key]);
+    }
+
+    document.body.appendChild(element);
+    const picker = new AmpDatePicker(element);
+
+    return {
+      element,
+      picker,
+    };
+  }
+
+  /**
+   * Create a template element
+   * @param {string} body
+   */
+  function createTemplate(body) {
+    const template = document.createElement('template');
+    template.setAttribute('type', 'amp-mustache');
+    template.content.appendChild(document.createTextNode(body));
+    return template;
+  }
+
+  const DEFAULT_TEMPLATE_ATTRS = {
+    'date-template': '',
+  };
+
+  /**
+   * Create a template from a body and attributes
+   * @param {string} body
+   * @param {!Object<string, string>=} opt_attrs
+   */
+  function createDateTemplate(body, opt_attrs) {
+    const template = createTemplate(body);
+    const attrs = Object.assign({}, DEFAULT_TEMPLATE_ATTRS, opt_attrs);
+    for (const key in attrs) {
+      template.setAttribute(key, attrs[key]);
+    }
+    return template;
+  }
+
+  beforeEach(() => {
+    win = env.win;
+    document = env.win.document;
+    toggleExperiment(win, 'amp-date-picker', true);
+  });
+
+  it('should render in the simplest case', () => {
+    const {picker} = createDatePicker({
+      layout: 'fixed-height',
+      height: 360,
+    });
+
+    return picker.layoutCallback().then(() => {
+      const container = picker.container_;
+      expect(container.children.length).to.be.greaterThan(0);
+    });
+  });
+
+  describe('getFormattedDate_', () => {
+    it('should render dates in the default format', () => {
+      const {picker} = createDatePicker({locale: 'en'});
+      const date = moment('2018-01-01');
+      const formattedDate = picker.getFormattedDate_(date);
+      expect(formattedDate).to.equal('2018-01-01');
+    });
+
+    it('should always render Unix epoch seconds in english digits', () => {
```

Comment:
These tests actually fail. Travis thinks that your numbers are wrong.

Reply:
👍 Just updated to include a timezone with the initialized `moment`

Reply:
Race conditions on GH lol.